lib/dry/component/injector.rb

```diff
@@ -27,37 +25,24 @@ def [](*deps)
         injector[*deps]
       end
 
-      # @api public
-      def args
-        strategies[:args]
-      end
-
-      # @api public
-      def hash
-        strategies[:hash]
+      def method_missing(name, *args, &block)
+        ::Dry::Component::Injector.new(container, options: options, strategy: name)
       end
 
-      # @api public
-      def kwargs
-        strategies[:kwargs]
+      def respond_to_missing?(name, _include_private = false)
+        injector.respond_to?(name)
```

Comment:
I've realised that this should actually just be `#respond_to?` inside a `BasicObject` subclass (since it doesn't actually offer `#respond_to?` by default). Will push up this adjustment shortly.